File: CCCC天梯赛/17年天梯决赛真题/L1-6.java
import java.math.BigInteger;
import java.util.*;
	
public class Main {
	static final int INF = 0x3f3f3f3f;
	
	public static void main(String[] args) {
		Scanner cin = new Scanner(System.in);
		
		int x;
		while(cin.hasNext()) {
			x = cin.nextInt();
			BigInteger xbi = new BigInteger(Integer.toString(x));
			BigInteger tmp = new BigInteger("1");
			int n = 1;
			while(true) {
				if(tmp.mod(xbi).intValue() == 0) {
					System.out.printf("%s %d", tmp.divide(xbi).toString(), n);
					break;
				}
				tmp = tmp.multiply(new BigInteger("10"));
				tmp = tmp.add(new BigInteger("1"));
				n++;
			}
		}
	}
}

File: CCCC天梯赛/17年天梯决赛真题/L1-6.cpp
#include <cstdio>
#include <cstring>
#include <iostream>
#include <algorithm>

using namespace std;

int main()
{
	long long x, s, n;
	while(cin>>x){
		n = 1;
		for(long long tmp = 1; ; tmp = 10*tmp+1){
			if(tmp % x == 0){
				cout<<tmp/x<<" "<<n<<endl;
				break;
			}
			n++;
		}
	}

	return 0;
}
